whole-tale/ngx-dashboard
# feat: view Tale Instance logs

## Problem
There is no way to view the logs for a running Tale Instance from the WholeTale Dashboard.

Fixes #308

## Approach
* Add a call to `/instance/{id}/log` within the `InstanceService`
* Abstract `view-logs-dialog` to work not only with a list of Job ids, but also with an instance id
    * if given an `Instance` resource as `this.data.instance`, the dialog will show the logs for that Instance
    * otherwise, it will assume the previous behavior and will attempt to read from `this.data.jobIds` to fetch/merge those logs accordingly

<img width="300" alt="Screen Shot 2022-12-05 at 2 17 50 PM" src="https://user-images.githubusercontent.com/1413653/205734720-22e758fd-f54a-4f1e-8757-85bc9eb8ce5e.png">

<img width="1227" alt="Screen Shot 2022-12-05 at 2 15 41 PM" src="https://user-images.githubusercontent.com/1413653/205734333-bac3bca7-97f5-4943-bae5-12c11eda4c2d.png">

## How to Test
Prerequisites: pull from upstream for all source repos, at least one Tale created/launched

1. Checkout this branch locally, rebuild the dashboard
2. Login to the WholeTale Dashboard
3. Navigate to the Tale Dashboard
4. Choose a Tale that your own, hover over it, and click View (or click on the Tale title)
    * You should be brought to the Run Tale page
5. Start the Tale, if you haven't already done so
6. Expand the Tale dropdown menu at the top-right
    * You should see a new option here to "View Logs"
7. Click the "View Logs" button
    * You should see the familiar "View Logs" dialog, but now reading the logs for the current Instance of the chosen Tale
